Src/WitsmlExplorer.Frontend/components/QueryEditorWidgetUtils.tsx

```diff
@@ -0,0 +1,113 @@
+import { Icon } from "@equinor/eds-core-react";
+import { createRoot } from "react-dom/client";
+import styled from "styled-components";
+import { getTag } from "./QueryEditorUtils";
+
+/**
+ * Handles the click event for opening a tag in the QueryEditor.
+ * Replaces the self-closing tag with an opening and closing tag and moves the cursor between the tags.
+ * @param editor The Ace editor instance.
+ * @param row The row number of the tag to open.
+ */
+const onClickOpenTag = (editor: any, row: number) => {
+  const session = editor.getSession();
+  const lineText = session.getLine(row);
+  const tag = getTag(lineText);
+  const newText = `${lineText.replace(/(\s*)\/>/, ">")}</${tag ?? ""}>`;
+  session.replace({ start: { row, column: 0 }, end: { row, column: lineText.length } }, newText);
+  editor.execCommand({
+    exec: () => {
+      editor.selection.moveCursorBy(0, -(tag?.length ?? 0) - 3);
+      editor.selection.clearSelection();
+    }
+  });
+};
+
+let updateTimeout: NodeJS.Timeout | null = null;
+
+/**
+ * Uses a deferred call to updateWidgets to avoid adding multiple icons to the same line when scrolling fast.
+ * @param editor The Ace editor instance.
+ * @param renderer The Ace renderer instance.
+ */
+export const updateLinesWithWidgets = (editor: any, renderer: any) => {
```

Comment:
Types: `editor: Ace.Editor, renderer: Ace.VirtualRenderer`

Reply:
We had a discussion and decided to skip this task because it was complicated due to issues with the package.